pkg/kubelet/kubelet_pods.go

```diff
@@ -387,8 +387,8 @@ func (kl *Kubelet) getServiceEnvVarMap(ns string) (map[string]string, error) {
 	// project the services in namespace ns onto the master services
 	for i := range services {
 		service := services[i]
-		// ignore services where ClusterIP is "None" or empty
-		if !v1helper.IsServiceIPSet(service) {
+		// ignore services where ClusterIP is "None" or empty but always add the master service
+		if !v1helper.IsServiceIPSet(service) && !(service.Namespace == kl.masterServiceNamespace && service.Name == masterServiceName) {
```

Comment:
I'm not a fan of hard-coding a special case here.

The fact that kubelet handling is involved means a version skewed kubelet wouldn't be able to deal with an external apiserver `kubernetes` service